src/shogun/io/serialization/BitseryVisitor.h

```diff
@@ -74,7 +74,13 @@ namespace shogun
 				}
 				void on(float64_t* v) override
 				{
-					m_s.value8b(*v);
+					if (static_cast<T*>(this)->m_auto_value.has_value())
+					{
+						*v = *static_cast<T*>(this)->m_auto_value;
+						static_cast<T*>(this)->m_auto_value.reset();
+					}
+					else
+						m_s.value8b(*v);
```

Comment:
@vigsterkr I really don't like this but I need a way to read an optional value.. Basically I store "auto", without initialisation, as a nullptr (like SGObject). If the value is nullptr all good, if not then I need to store somehow so that the visitor goes through it (to avoid repositioning the buffer). If you checkout in any.h how I implemented the visitor for std::variant to be used in auto init you'll see why I need this foobarness. I am sure there is a better way, but I can't think of it... Any ideas?